file_store/src/file_sink.rs

```diff
@@ -274,8 +301,19 @@ impl FileSink {
                             }
                         };
                         let _ = on_write_tx.send(res);
-                    },
-                    Some(Message::Flush) => self.flush().await?,
+                    }
+                    Some(Message::Flush(on_flush_tx)) => {
+                        let res = self.flush().await;
+                        let _ = on_flush_tx.send(res);
+                    }
+                    Some(Message::FetchManifest(on_fetch_tx)) => {
+                        let _ = self.flush().await;
+                        if let Some(ref mut manifest) = self.manifest {
+                            let _ = on_fetch_tx.send(Ok(mem::take(manifest)));
```

Comment:
NIT: Can you use `manifest.take()` instead?

Reply:
Nope - `take` doesn't do the same thing as `Option::take` for Vecs